x/finality/keeper/power_dist_change.go

```diff
@@ -326,6 +350,20 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 		}
 	}
 
+	k.processBtcDelegations(ctx, cacheFpByBtcPkHex, newActiveBtcDels, func(fp, del sdk.AccAddress, sats uint64) {
```

Comment:
Can we do this inside the `for _, event := range events` loop? This loop was supposed to be the "event bus" that dispatches notifications to relevant modules, including `x/incentive`

Reply:
we could if the events were sorted and never received more than once 

ref https://github.com/babylonlabs-io/babylon/pull/306#discussion_r1886719715

One BTC delegation can generate 2 unbonding events

Reply:
I see. Probably a better fix is then distinguishing between two types of unbonded events, but this is certainly out of this PR's scope.

Also, why is `newActiveBtcDels` needed given that a BTC del cannot be active twice?

Reply:
I didn't had in mind that the events were deterministic, my bad 

I can process the reward tracker right at the moment the event are received, I will update tomorrow and remove the `newActiveBtcDels` and `newUnbondedBtcDels`

Reply:
removed